go/arrow/flight/flightsql/driver/config.go

```diff
@@ -19,9 +19,53 @@ import (
 	"crypto/tls"
 	"fmt"
 	"net/url"
+	"sync"
 	"time"
+
+	"github.com/google/uuid"
+)
+
+// TLS configuration registry
+var (
+	tlsConfigRegistry = map[string]*tls.Config{
+		"skip-verify": {InsecureSkipVerify: true},
+	}
+	tlsRegistryMutex sync.Mutex
```

Comment:
would it make sense to use an `RWMutex` so the `GetTLSConfig` function can use `RLock` and `RUnlock` ?

Reply:
I can certainly do this, but I don't expect this to be in a hot-path. Remember, the mutex is locked if you register a config _and_ when you are calling `Open` (which in turn does the DSN parsing). Do you want me to use a `RWMute`?

Reply:
I agree that it's not likely in a hot path and unlikely to be a cause for contention. It's probably alright for it to be a regular mutex. this is fine.